bot/exts/moderation/infraction/_utils.py

```diff
@@ -298,3 +300,26 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    confirmed = False
+
+    @discord.ui.button(label="Confirm", style=ButtonStyle.red)
+    async def confirm(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "confirm" button is pressed."""
+        self.confirmed = True
+        await interaction.response.defer()
+        self.stop()
+
+    @discord.ui.button(label="Cancel", style=ButtonStyle.green)
+    async def cancel(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "cancel" button is pressed."""
+        await interaction.response.send_message("Cancelled infraction.")
+        self.stop()
+
+    async def on_timeout(self) -> None:
```

Comment:
Good place to use the `typing.override` decorator

https://docs.python.org/3/library/typing.html#typing.override

Reply:
We're not on 3.12 yet, so can't use it yet (we were blocked until very recently waiting for dependencies to update).